src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -1482,8 +1485,263 @@ class OPENASSETIO_CORE_EXPORT Manager final {
+   * references related to the input entity reference, or an error.
+   *
+   * Otherwise, the result is populated with an error object detailing
+   * the reason for the failure to fetch the related entities.
+   *
+   * Errors that are not specific to the entity relationship will be
+   * thrown as an exception.
+   *
+   * @param entityReference An @ref entity_reference to query
+   * the specified relationship for.
+   *
+   * @param relationshipTraitsData The traits of the relationship to
+   * query.
+   *
+   * @param pageSize The size of each page of data. The page size is
+   * fixed for the lifetime of pager object given to the @p
+   * successCallback. Must be greater than zero.
+   *
+   * @param relationsAccess The intended usage of the returned
+   * references.
+   *
+   * @param context The calling context.
+   *
+   * @param resultTraitSet A hint as to what traits the returned
+   * entities should have.
+   *
+   * @param errorPolicyTag  Parameter for selecting the appropriate
+   * overload (tagged dispatch idiom). See @ref
+   * BatchElementErrorPolicyTag::Variant.
+   *
+   * @return Object containing either the EntityReferencePager pointer,
+   * which pages over an unbounded set of entity references related to
+   * the input entity reference, or an error.
+   *
+   * @throws errors.InputValidationException if @p pageSize is zero.
+   *
+   * @throws errors.NotImplementedException Thrown when this method is
+   * not implemented by the manager. Check that this method is
+   * implemented before use by calling @ref hasCapability with @ref
+   * Capability.kRelationshipQueries.
+   *
+   * @see @ref Capability.kRelationshipQueries
+   */
+  std::variant<errors::BatchElementError, EntityReferencePagerPtr> getWithRelationship(
+      const EntityReference& entityReference, const trait::TraitsDataPtr& relationshipTraitsData,
+      size_t pageSize, access::RelationsAccess relationsAccess, const ContextConstPtr& context,
+      const trait::TraitSet& resultTraitSet,
+      const BatchElementErrorPolicyTag::Variant& errorPolicyTag);
+
+  /**
+   * Query for entity references that are related to the input
+   * references by the relationship defined by a set of traits and
+   * their properties.
+   *
+   * See documentation for the <!--
+   * --> @ref getWithRelationship(const EntityReferences&, <!--
+   * --> const trait::TraitsDataPtr&, size_t, <!--
+   * --> access::RelationsAccess, const ContextConstPtr&, <!--
+   * --> const RelationshipQuerySuccessCallback&, <!--
+   * --> const BatchElementErrorCallback& errorCallback, <!--
+   * --> const trait::TraitSet&)
+   * "callback variation" for more details on relationship behaviour.
+   *
+   * Any errors that occur during resolution will be immediately thrown
+   * as an exception, either from the @ref manager plugin (for errors
+   * not specific to the entity reference) or as a
+   * @fqref{errors.BatchElementException}
+   * "BatchElementException"-derived error.
```

Comment:
[fix] copy-paste  error from `resolve`

Reply:
Can you be more specific I don't understand what's specific to resolve here. You do get a BatchElementException derived error as far as I know?

Reply:
I think I meant the "Any errors that occur during resolution" bit, in combination with the "not specific to the entity reference" bit, which is a copy-paste from the `resolve` docs.  Also applies to other signatures, where I've commented variously and inconsistently.

To be fair, it's reasonable to view the word "resolution" as a more generic concept, since we're "resolving the relationships", so we could keep that phrasing. Still, like in other comments, worth swapping out "entity reference" for "relationship".

Reply:
Ah but then we've changed a bunch of others re. "relationship" vs. "entity reference" - should really be consistent.

Reply:
We did this together, David gave verbal consent for rebase and merge. Resolving this.